### PR TITLE
Chore/UI changes

### PR DESCRIPTION
**Pull request by Ballo1988**
- Made changes to meet clients updates
- Generated new migration with required fields
- Removed alergies from pain form since it was repeated
- Updated controller and model

SCREENSHOTS
image
image

### CODE REVIEW

File: app/models/patient.rb [hunk 1]
```diff
@@ -8,14 +8,14 @@ class Patient < ApplicationRecord
                     tsearch: { prefix: true }
                   }
 
-  enum gender: { masculino: 0, femenino: 1 }
-  enum marital_status: { casado: 0, divorciado: 1, soltero: 2, union_libre: 3, viudo: 4 }
+  enum gender: { m: 0, f: 1 }
```
Comment (itsmeurbi) [inline app/models/patient.rb]: Actually it is ok having the whole symbol. This line will generate one helper per symbol. And having thins like this just make no sence:
```
braulio = Patient.find(1)
braulio.m?
braulio.f?
braulio.s?
braulio.c?
braulio.u_l?
```

Comment (itsmeurbi) [reply app/models/patient.rb]: Let's roll this back is it was. If you need some particular text on the views, add the required things on a helper or directly on the html tags

**Review by itsmeurbi — CHANGES_REQUESTED** [review]

File: app/models/patient.rb (hunk 1 above)
Comment (Ballo1988) [reply app/models/patient.rb]: Reversed the changes, Should I edit the helpers to only show M, F, etc. or just display the whole word?

Comment (itsmeurbi) [reply app/models/patient.rb]: Document says it should only be `M` and `F`, so answers is yes

Comment (itsmeurbi) [reply app/models/patient.rb]: Same with the `marital_status`

File: app/controllers/patients_controller.rb [hunk 2]
```diff
@@ -63,6 +65,15 @@ def destroy
 
   private
 
+  def selected_values
```
Comment (itsmeurbi) [inline app/controllers/patients_controller.rb]: Nope

File: app/views/patients/_form.html.erb [hunk 3]
```diff
@@ -47,75 +28,105 @@
...
-        <%= f.label :occupations, 'Ocupaciones:' %>
-        <%= f.text_field :occupations, placeholder: 'Deporte, Danza, Lectura...',
-                                       class: 'w-full rounded p-2 text-gray-900 border border-gray-300 focus:ring-blue-500 focus:border-blue-500' %>
+        <div>
+          <%= f.label :gender, 'Sexo:' %>
+          <%= f.select :gender, options_for_select(gender_select_options, selected: @gender), {}, {class: 'w-full rounded p-2 text-gray-900 border border-gray-300 focus:ring-blue-500 focus:border-blue-500'} %>
+        </div>
+        <div>
+          <%= f.label :marital_status, 'Estado Civil:' %>
+          <%= f.select :marital_status, options_for_select(marital_status_select_options, selected: @marital_status), {}, {class: 'w-full rounded p-2 text-gray-900 border border-gray-300 focus:ring-blue-500 focus:border-blue-500'} %>
```
Comment (itsmeurbi) [inline app/views/patients/_form.html.erb]: In all this selects you already have access to the patient instance, just ask for the value on each attribute
There's no need to create anda carry instance variables all over the template
```suggestion
          <%= f.select :marital_status, options_for_select(marital_status_select_options, selected: @patient.marital_status), {}, {class: 'w-full rounded p-2 text-gray-900 border border-gray-300 focus:ring-blue-500 focus:border-blue-500'} %>  
```

File: app/helpers/patient_helper.rb [hunk 4]
```diff
@@ -2,11 +2,11 @@
 
 module PatientHelper
   def gender_select_options
-    Patient.genders.map { |k, _v| [k.titleize, k] }
+    Patient.genders.map { |k, _v| [k.to_s[/\b\w/].titleize, k] }
```
Comment (itsmeurbi) [inline app/helpers/patient_helper.rb]:
```suggestion
    Patient.genders.map { |k, _v| [k.to_s.first.upcase, k] }
```

File: app/helpers/patient_helper.rb [hunk 5]
```diff
@@ -2,11 +2,11 @@
...
   end
 
   def marital_status_select_options
-    Patient.marital_statuses.map { |k, _v| [k.titleize, k] }
+    Patient.marital_statuses.map { |k, _v| [k.to_s[/\b\w/].titleize, k] }
```
Comment (itsmeurbi) [inline app/helpers/patient_helper.rb]:
```suggestion
    Patient.marital_statuses.map { |k, _v| [k.to_s.first.upcase, k] }
```

**Review by itsmeurbi — CHANGES_REQUESTED** [review]

File: app/controllers/patients_controller.rb (hunk 2 above)
Comment (Ballo1988) [reply app/controllers/patients_controller.rb]: Fixed

**Review by itsmeurbi — APPROVED** [review]

**Pull request merged**